resolver.go

```diff
@@ -500,6 +500,11 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 			break
 		}
 		if resp == nil {
+			resp = new(dns.Msg)
+			resp.SetRcode(query, dns.RcodeServerFailure)
+			if err = w.WriteMsg(resp); err != nil {
+				logrus.Errorf("error writing resolver resp after failure, %s", err)
```

Comment:
For consistency, I suggest using the `[resolver] ` prefix here as well. Perhaps something like:

```go
logrus.Errorf("[resolver] failed to write error response: %s", err)
```